packages/backend/src/clients/dbtCloud/transformer.ts

```diff
@@ -109,11 +109,8 @@ export const dbtCloudTransfomers: SemanticLayerTransformer<
     semanticLayerQueryToQuery: (query) => {
         const { metrics, dimensions, timeDimensions } = query;
         return {
-            metrics: metrics.map((metric) => ({ name: metric })),
-            groupBy: [
-                ...dimensions.map((dimension) => ({ name: dimension })),
-                ...timeDimensions,
-            ],
+            metrics,
+            groupBy: [...dimensions, ...timeDimensions],
```

Comment:
One thing here, we should map `timeDimensions` to pass `granularity` as `grain` to `groupBy`: `{name: 'a', granularity: 'day'}` => `{name: 'a', grain: 'day'}`. Because the dbt type nullable then TS is not complaining